File: wesender_client.py
import requests
import logging
from typing import Dict, Any, Optional

logger = logging.getLogger(__name__)

class WeSenderClient:
    """
    Client for WaSenderAPI (WhatsApp).
    Docs: https://wasenderapi.com
    
    Endpoint: POST https://wasenderapi.com/api/send-message
    Auth: Bearer token
    Payload: { "to": "+1234567890", "text": "Hello!" }
    """
    
    DEFAULT_API_URL = "https://wasenderapi.com/api/send-message"

    def __init__(self, api_key: str, api_url: Optional[str] = None):
        self.api_key = api_key
        self.api_url = api_url.rstrip('/') if api_url else self.DEFAULT_API_URL
        
        self.headers = {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json",
            "Accept": "application/json"
        }

    def send_message(self, phone: str, message: str) -> Dict[str, Any]:
        """
        Send a WhatsApp text message.
        
        Args:
            phone: Recipient phone number with international code (e.g. +393331234567).
            message: The text message content.
            
        Returns:
            Dict with success status and API response data.
        """
        payload = {
            "to": phone,
            "text": message
        }
        
        try:
            logger.info(f"Sending WhatsApp message to {phone} via WaSenderAPI...")
            response = requests.post(
                self.api_url, 
                json=payload, 
                headers=self.headers, 
                timeout=15
            )
            response.raise_for_status()
            
            return {
                "success": True, 
                "data": response.json() if response.content else {},
                "status_code": response.status_code
            }
            
        except requests.RequestException as e:
            logger.error(f"WaSenderAPI Error: {str(e)}")
            error_details = {"error": str(e)}
            if hasattr(e, 'response') and e.response is not None:
                error_details["status_code"] = e.response.status_code
                try:
                    error_details["response_text"] = e.response.text
                except:
                    pass
            return {"success": False, "details": error_details}

File: api/wesender_client.php
<?php
/**
 * Floormad Workflow - WeSender Client (WhatsApp via WaSenderAPI)
 */
require_once __DIR__ . '/config.php';

class WeSenderClient
{
    private $api_key;
    private $api_url;

    const DEFAULT_API_URL = 'https://wasenderapi.com/api/send-message';

    public function __construct($api_key, $api_url = null)
    {
        $this->api_key = $api_key;
        $this->api_url = $api_url ? rtrim($api_url, '/') : self::DEFAULT_API_URL;
    }

    /**
     * Send a WhatsApp text message.
     */
    public function send_message($phone, $message)
    {
        $payload = [
            'to' => $phone,
            'text' => $message
        ];

        $ch = curl_init($this->api_url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_POST, true);
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($payload));
        curl_setopt($ch, CURLOPT_TIMEOUT, 15);
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            'Authorization: Bearer ' . $this->api_key,
            'Content-Type: application/json',
            'Accept: application/json'
        ]);

        $response = curl_exec($ch);
        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
        $error = curl_error($ch);
        curl_close($ch);

        if ($error) {
            error_log("WaSenderAPI Error: $error");
            return ['success' => false, 'details' => ['error' => $error]];
        }

        if ($httpCode >= 400) {
            error_log("WaSenderAPI HTTP Error: $httpCode - $response");
            return [
                'success' => false,
                'details' => [
                    'error' => "HTTP $httpCode",
                    'status_code' => $httpCode,
                    'response_text' => $response
                ]
            ];
        }

        $decoded = json_decode($response, true);
        return [
            'success' => true,
            'data' => $decoded ?: [],
            'status_code' => $httpCode
        ];
    }
}
